### PR TITLE
Add a dedicated Golden update workflow

### PR DESCRIPTION
**Pull request by Abestanis**
This adds a dedicated workflow that can be triggered from the Actions tab on GitHub to update the Golden test artifacts on a branch. This required splitting the existing workflow for reusabillity. The workflow uploads the updated test artifacts, but it can also commit the changes directly to the repository:
Update Goldens Workflow
Update Goldens Workflow 2
image

I had to make this pull request against `master`, because the default branch must have the workflow file, otherwise it won't show up in the Actions tab. To use this workflow, the branch must also have the workflow file, so it would be nice if we could merge `master` into `1.0.8` after this is merged.

To prevent just everyone from being able to trigger golden updates and auto commits, I added a check a commit was made on the branch under the same name as the username of the user running the workflow. This should work, as long as people have the same git and GitHub username. We could also prevent auto commits on master.

### CODE REVIEW

File: .github/workflows/update_goldens.yml
Comment (nt4f04uNd) [inline .github/workflows/update_goldens.yml:9]: unnecessary "the"

File: .github/workflows/update_goldens.yml
Comment (nt4f04uNd) [inline .github/workflows/update_goldens.yml:24]: let's default these two parameters into the script

File: .github/workflows/update_goldens.yml
Comment (nt4f04uNd) [inline .github/workflows/update_goldens.yml:36]: This doesn't verify actual permissions - for example you don't have permissions to commit into any of the branches, but you have commits with your signature in the tree.

I would honestly prefer that we rather just limit this workflow to repository admins, so I as reviewer can trigger it on a pull request before I merge, after I reviewed all the goldens manually. It will regenerate all goldens and it's not ideal but I think it's an okay policy and would avoid potential flakes that could happen because of tolerance gap.

**Review by nt4f04uNd — COMMENTED** [review]
This is a cool feature

File: .github/workflows/update_goldens.yml
Comment (nt4f04uNd) [reply .github/workflows/update_goldens.yml:36]: By admins I meant collaborators, actually

Comment (Abestanis) [reply .github/workflows/update_goldens.yml:34]: > This doesn't verify actual permissions

Yes, I mentioned this in the PR description briefly.This was the easiest way that I could think of that allows people to commit to their own branches or "collaborators" which have previously committed. I was hesitant to use the [actual collaborators list from the GitHub repository](https://github.com/nt4f04uNd/sweyer/graphs/contributors), because it is empty, since you squash all commits on a merge request, so GitHub counts the code as your contribution.

I will look into an alternative solution, because it would be nice if someone at least could commit into their own branches.

Comment (nt4f04uNd) [reply .github/workflows/update_goldens.yml:34]: > I was hesitant to use the [actual collaborators list from the GitHub repository](https://github.com/nt4f04uNd/sweyer/graphs/contributors), because it is empty, since you squash all commits on a merge request, so GitHub counts the code as your contribution.

Github doesn't count squashed commits as my contribution, since they are signed with the email of the pull request author, which is you. I might be co-author of the squshed commit, but that is only if I commited to pull request branch.

This list also doesn't display who is a collaborator on the repository, this is a list of all contributors.
Also, Github counts contibutions only on master branches of non-fork repositories, hence it doesn't display you in the list, until https://github.com/nt4f04uNd/sweyer/commits/1.0.8 is merged into master.

Comment (Abestanis) [reply .github/workflows/update_goldens.yml:34]: Ok, thanks for the clarification, I just assumed that the list would be empty based on what I could see. I would still like to try to allow people to commit to their own branches, its just very convenient. How about this:

We can check if someone is a contributor using [`actions-cool
/check-user-permission`](https://github.com/actions-cool/check-user-permission)
If the branch has a parent, we allow everyone who has made a commit on that branch up to it's parent, meaning we are looking just at the new commits on the branch.

Would that be ok with you?

Comment (Abestanis) [reply .github/workflows/update_goldens.yml:34]: Actually, I wonder if checking for the write user-permission is enough...

Comment (Abestanis) [reply .github/workflows/update_goldens.yml]: @nt4f04uNd could you please test if you can run the workflow with auto commit against `master` in [my fork](https://github.com/Abestanis/sweyer/actions/workflows/update_goldens.yml)? Also, try to open a pull request and check the auto commit against that branch.

Comment (Abestanis) [reply .github/workflows/update_goldens.yml]: Hang on, it's not working just yet.

Comment (Abestanis) [reply .github/workflows/update_goldens.yml]: Ok, you can try now

Comment (nt4f04uNd) [reply .github/workflows/update_goldens.yml]: > could you please test if you can run the workflow with auto commit against master in [my fork](https://github.com/Abestanis/sweyer/actions/workflows/update_goldens.yml)?

I cannot, there is no button to run a workflow for me.

I see that you try to disable only the step that actually commits goldens, but I would prefer to make this entire workflow only accessible to collaborators, it otherwise incentivizes to run it for no reason - reviewers should decide if new goldens are good or not.

Comment (nt4f04uNd) [reply .github/workflows/update_goldens.yml]: Ah, nevermind, workflow_dispatch is by default only accessible to collaborators

File: .github/workflows/update_goldens.yml
Comment (nt4f04uNd) [inline .github/workflows/update_goldens.yml:39]: The workflow should fail if it doesn't have permissions to commit

So it's better to remove this entire thing witrh `permissionCheck`

**Review by nt4f04uNd — APPROVED** [review]
LGTM, except one comment

File: .github/workflows/update_goldens.yml
Comment (Abestanis) [reply .github/workflows/update_goldens.yml:39]: Ok, I think it is actually perfect that you can only run the workflow as a collaborator, because you can always run it on your own local fork. No need to run it here.

I will update the contributing documentation accordingly. Please merge this into `1.0.8` as well afterwards.

**Comment by nt4f04uNd** [comment]
Oh I just realized that I would not be able to trigger this on forked branches, right? That would be better if I didn't have to run it on master... Any ideas?

**Comment by Abestanis** [comment]
> Oh I just realized that I would not be able to trigger this on forked branches, right?

Yes, but the owner of the fork can.

**Pull request merged**
